pkg/coastguard/main.go

```diff
@@ -0,0 +1,38 @@
+package main
+
+import (
+	"flag"
+	"sync"
+
+	"github.com/submariner-io/submariner/pkg/signals"
+	"k8s.io/klog"
+
+	"github.com/submariner-io/coastguard/pkg/clusters/discovery"
+	"github.com/submariner-io/coastguard/pkg/controller"
+)
+
+func main() {
+	klog.InitFlags(nil)
+	flag.Parse()
+
+	klog.V(2).Info("Starting coastguard-network-policy-sync")
+
+	// set up signals so we handle the first shutdown signal gracefully
+	stopCh := signals.SetupSignalHandler()
+
+	var wg sync.WaitGroup
+
+	wg.Add(1)
```

Comment:
No, I want to wait until the coastguard Controller exited , not until the channel closed. What if the channel closed but the controller didn't finish something it had to finish?

golang will kill the exiting goroutines as soon as main exits, example:

```
package main

import (
	"fmt"
	"time"
)

func routine(n int) {
	for i := 0; i < 100; i++ {
		fmt.Printf("go routine %d at %d\n", n, i)
		time.Sleep(time.Second * 2)
	}
}

func main() {
	for i := 0; i < 5; i++ {
		go routine(i)
	}
	time.Sleep(2 * time.Second)

}
```
```

[ajo@rhvirt-casa test]$ go run main.go 
go routine 0 at 0
go routine 1 at 0
go routine 4 at 0
go routine 2 at 0
go routine 3 at 0
go routine 0 at 1
go routine 2 at 1
go routine 3 at 1

```

Reply:
@mpeterson is right in this case - a chan could be used since there's only 1 thing to wait on:

```
runStopped := make(chan struct{})
go func() {
    defer close(runStopped)
    coastGuardController.Run(stopCh)
}()

<-runStopped
```
It essentially amounts to the same thing as using a `WaitGroup` but `WaitGroup` seems to be the common pattern in this case from what I've seen elsewhere. If you have more than 1 thing to wait on then you definitely need a `WaitGroup` (unless you use multiple channels which would be cumbersome).

Reply:
True, using a separate channel would work, that's right :)

I thought he was suggesting to use the same stopCh as the other processes

:+1: